lib/routes/shared/security_pin/lock_screen.dart

```diff
@@ -43,50 +40,27 @@ class _AppLockScreenState extends State<AppLockScreen> {
                 ),
                 elevation: 0.0,
               )
-            : null, 
+            : null,
         body: PinCodeWidget(
           _label,
-          _enteredPinCode,
           widget.canCancel,
-          _errorMessage,
-          (numberText) => _onNumButtonPressed(numberText),
-          (enteredPinCode) => _setPinCodeInput(enteredPinCode),
+          (enteredPinCode) => _onPinEntered(enteredPinCode),
+          key: _key,
         ),
       ),
     );
   }
 
-  _onNumButtonPressed(String numberText) {
-    _setErrorMessage("");
-
-    if (_enteredPinCode.length < PIN_CODE_LENGTH) {
-      _setPinCodeInput(_enteredPinCode + numberText);
-    }
-    if (_enteredPinCode.length == PIN_CODE_LENGTH) {
-      Future.delayed(Duration(milliseconds: 200), () {
-        if (_enteredPinCode == widget.securityModel.pinCode) {
-          if (widget.onUnlock != null) {
-            widget.onUnlock();
-            return;
-          }
-          Navigator.pop(context, true);
-        } else {
-          _setPinCodeInput("");
-          _setErrorMessage("Incorrect PIN");
-        }
-      });
+  _onPinEntered(String enteredPinCode) {
+    if (enteredPinCode == widget.securityModel.pinCode) {
+      if (widget.onUnlock != null) {
+        widget.onUnlock();
+        return;
+      }
+      Navigator.pop(context, true);
+    } else {
+      _key.currentState.setPinCodeInput("");
```

Comment:
Can we implement the clear input inside the `PinCodeWidget` ? In a way that we clear the value every time before calling `onPinEntered`. It will save us maintaining this in all widgets.